docs/model_architecture.md

```diff
@@ -0,0 +1,93 @@
+# Model Architecture
+
+---
+
+## Table of Contents
+
+ - [Main classes](#main-classes) 
+ - [Additional Classes](#additional-classes)
+ - [Extensions](#extensions)
+ - [All modules, automated documentation](#all-modules-automated-documentation)
+
+
+---
+
+
+
+## Main classes
+
+We have already mentioned the hierarchy of the four most important classes which, ffrom the point of view of function, could be expressed as: `Model` > `Simulation` > `Experiment` > `Engine`.
+
+* `Model` class expresses the model specification, most important part of which is passed throught the parameters `initial_state`, `state_update_blocks=state_update_blocks`, `params`. Below is an example of an instance from the Model class:
+
+```python
+model = Model(initial_state=initial_state, 
+              state_update_blocks=state_update_blocks, 
+              params=params)
+```
+
+* `Simulation` is a class used for simulating a `Model`. It extends the class `Executable` and contains nan implemetnation of the method `run`, most other elements are inherited from the class `Executable`. Example of initiating the simulation:
+
+```python
+simulation = Simulation(model=model, timesteps=100_000, runs=1)
+
+result = simulation.run()
+df = pd.DataFrame(result)
+```
+
+* `Experiment` consists of one or more `Simulation` snd is sldo an extension of the class `Executable`. Example of initiating an experiment consisting of one or more of `Simulation`:
+
+```python
+experiment = Experiment([simulation_1, simulation_2, simulation_3])
+result = experiment.run()
+
+df = pd.DataFrame(result)
+```
+
+* `Engine` - class that handles configuration and execution of experiments and simulations. It is a method first defined for the class `Executable` and then inherited by classes  `Simulation` and `Experiment`.
+
+
+## Additional Classes
+
+### Configuration
+
+* `SimulationExecutionSpecification` - a data class that prepares some of the elements of simulation specification that are common for most experiments, in particular the organisation into blocks, steps and substeps.
+* `SimulationExecution` - extends `SimulationExecutionSpecification` and defines various properties and functions needed to execute the simulation.
+* `Executable` - instances of class `Executable` contain all information about the model specification and simulation configuration and can be passed to either `Simulation` (for a single simulation) or to `Experiment` (for multiple simulations). `E` contains a method `run` which is not implemented, as that implementation takes place at the level of either `Simulation` or `Experiment`.
```

Comment:
What is `E`? I think these descriptions are probably not clear enough to include in the documentation just yet, might suggest we remove these for the moment.